Source/Controllers/MainViewControllers/TableRelations/SPTableRelations.m

```diff
@@ -254,14 +267,31 @@ - (IBAction)addRelation:(id)sender
 		[connection setEncoding:@"utf8mb4"];
 	}
 
-	// Get all InnoDB tables in the current database
-    SPMySQLResult *result = [connection queryString:[NSString stringWithFormat:@"SELECT table_name FROM information_schema.tables WHERE table_type = 'BASE TABLE' AND engine = 'InnoDB' AND table_schema = %@ ORDER BY table_name ASC", [[tableDocumentInstance database] tickQuotedString]]];
-    [result setDefaultRowReturnType:SPMySQLResultRowAsArray];
-    [result setReturnDataAsStrings:YES]; // TODO: Workaround for #2699/#2700
-    for (NSArray *eachRow in result) {
-        [refTablePopUpButton addItemWithTitle:[eachRow objectAtIndex:0]];
+    // Get all InnoDB Databases
+    
+    NSArray *theDatabaseList = [connection databases];
+    NSMutableArray *allDatabases;
+    allDatabases = [[NSMutableArray alloc] initWithCapacity:[theDatabaseList count]];
+    for (NSString *databaseName in theDatabaseList)
+    {
+        // If the database is either information_schema or mysql then it is classed as a
+        // system database; similarly, performance_schema in 5.5.3+ and sys in 5.7.7+
+        if (![databaseName isEqualToString:SPMySQLDatabase] &&
+            ![databaseName isEqualToString:SPMySQLInformationSchemaDatabase] &&
+            ![databaseName isEqualToString:SPMySQLPerformanceSchemaDatabase] &&
+            ![databaseName isEqualToString:SPMySQLSysDatabase]) {
+             [allDatabases addObject:databaseName];
+        }
+    }
+    if (allDatabases != nil) {
+        [refDatabasePopUpButton addItemsWithTitles:allDatabases];
```

Comment:
Should we empty out the items that may be there before adding new ones?

Reply:
it's done, line 259